packages/data/src/data-manager.ts

```diff
@@ -286,111 +289,126 @@ export class DataManager implements DataManagerInterface {
+        segmentsMatched = false;
+      if (visitorProperties) {
+        if (
+          Array.isArray(experience?.audiences) &&
+          experience.audiences.length
+        ) {
+          // Get attached transient and/or permnent audiences
+          audiences = this.getItemsByIds(
+            experience.audiences,
+            'audiences'
+          ) as Array<ConfigAudience>;
 
-            //if visitor already bucketed into this experience, check only audiences of type transient
-            audiencesToCheck = audiences.filter(
-              (audience) =>
-                !(isBucketed && audience.type === ConfigAudienceTypes.PERMANENT)
+          // If visitor already bucketed into this experience, check only audiences of type transient
+          audiencesToCheck = audiences.filter(
+            (audience) =>
+              !(isBucketed && audience.type === ConfigAudienceTypes.PERMANENT)
+          );
+          if (audiencesToCheck.length) {
+            // Validate visitorProperties against audiences rules
+            matchedAudiences = this.filterMatchedRecordsWithRule(
+              audiencesToCheck,
+              visitorProperties,
+              'audience',
+              identityField
             );
-            if (audiencesToCheck.length) {
-              // Validate visitorProperties against audiences rules
-              matchedAudiences = this.filterMatchedRecordsWithRule(
-                audiencesToCheck,
-                visitorProperties,
-                'audience',
-                identityField
-              );
-              // Return rule errors if present
-              matchedErrors = matchedAudiences.filter((match) =>
-                Object.values(RuleError).includes(match as RuleError)
-              );
-              if (matchedErrors.length) return matchedErrors[0] as RuleError;
-              if (matchedAudiences.length) {
-                for (const item of matchedAudiences) {
-                  this._loggerManager?.info?.(
-                    'DataManager.matchRulesByField()',
-                    MESSAGES.AUDIENCE_MATCH.replace('#', item?.[identityField])
-                  );
-                }
+            // Return rule errors if present
+            matchedErrors = matchedAudiences.filter((match) =>
+              Object.values(RuleError).includes(match as RuleError)
+            );
+            if (matchedErrors.length) return matchedErrors[0] as RuleError;
+            if (matchedAudiences.length) {
+              for (const item of matchedAudiences) {
+                this._loggerManager?.info?.(
+                  'DataManager.matchRulesByField()',
+                  MESSAGES.AUDIENCE_MATCH.replace('#', item?.[identityField])
+                );
               }
             }
+            audiencesMatched = Boolean(matchedAudiences.length);
           } else {
+            audiencesMatched = true; // Empty non-permanent experience audiences list means no restrictions
```

Comment:
why this non permanent stuff @abbaseya  ? who cares? up to you if you want to remove or not :) rest looks good

Reply:
just for debugging later, might help...